Release Notes for 1.9.4 (#9593)
istio/istio.io: content/en/news/releases/1.9.x/announcing-1.9.4/index.md

---
title: Announcing Istio 1.9.4
linktitle: 1.9.4
subtitle: Patch Release
description: Istio 1.9.4 patch release.
publishdate: 2020-04-27
release: 1.9.4
aliases:
    - /news/announcing-1.9.4
---

This release contains bug fixes to improve robustness. This release note describes what’s different between Istio 1.9.3 and Istio 1.9.4

{{< relnote >}}

## Changes

> - **Fixed** an issue where the Istio operator prunes the resource that doesn't belong to specific Istio operator CR. ([Issue #30833](https://github.com/istio/istio/issues/30833))

Reviewer: ```suggestion
- **Fixed** an issue where the Istio operator prunes any resources that don't belong to a specific Istio operator custom resource. ([Issue #30833](https://github.com/istio/istio/issues/30833))
```

I'm still not sure if this expresses the same behavior described here though: https://github.com/istio/istio/issues/30833#issuecomment-823728149

Reviewer: I think what I'm understanding from that comment in the PR is: 

```
- **Fixed** an issue where the Istio operator prunes all resources created by the operator, including itself. Now the operator will only remove resources belonging to the custom resource.
```

Reviewer: @morvencao can you please take a look?

Reviewer 2: Correct, the description looks good to me, with the fix, istio operator will only prune the resources for enabled components in the CR.

Author: Thanks

> - **Fixed** an issue where the Istio operator prunes any resources that don't belong to a specific Istio operator custom resource. ([Issue #30833](https://github.com/istio/istio/issues/30833))

Reviewer: As per the previous discussion:

```suggestion
- **Fixed** an issue where the Istio operator prunes all resources created by the operator, including itself. Now the operator will only remove resources belonging to the custom resource.
```

- **Fixed** an issue ensuring lease duration is always greater than the user configured `RENEW_DEADLINE` for Istio operator manager. ([Issue #27509](https://github.com/istio/istio/issues/27509))

- **Fixed** an issue where a certificate provisioned by sidecar proxy cannot be used by Prometheus. ([Issue #29919](https://github.com/istio/istio/issues/29919))

- **Fixed** an issue that creates an IOP under `istio-system` when installing Istio in another namespace. ([Issue #31517](https://github.com/istio/istio/issues/31517))

> - **Fixed** an issue when using `PeerAuthentication` to turn off mTLS while using multi-network, non-mTLS endpoints will be removed from the cross-network load-balancing endpoints to prevent 500 errors. ([Issue #28798](https://github.com/istio/istio/issues/28798))

Reviewer: ```suggestion
- **Fixed** an issue when using `PeerAuthentication` to turn off mTLS while using multi-network. Now non-mTLS endpoints will be removed from cross-network load-balancing endpoints to prevent 500 errors. ([Issue #28798](https://github.com/istio/istio/issues/28798))
```

Author: Thanks

> 

Reviewer 3: ```suggestion
- **Fixed** istiod never becoming ready when it fails to read resources from clusters configured via remote secrets.
After a timeout configured by `PILOT_REMOTE_CLUSTER_TIMEOUT` (default 30s), istiod will become ready without
syncing remote clusters. The stat `remote_cluster_sync_timeouts` will be incremented when this occurs.
  ([Issue #30838](https://github.com/istio/istio/issues/30838))

```

> 

Reviewer 3: ```suggestion

- **Fixed** an issue where Istiod will not create a self-signed root CA and istio-ca-root-cert configmap when `values.global.pilotCertProvider` is `kubernetes`. ([Issue 32023](https://github.com/istio/istio/issues/32023))

```

Author: Will add these too. Thanks for taking a look.

Author: Done.

Reviewer: Created https://github.com/istio/istio/pull/32485. I get the feeling that a lot of developers still aren't sure when to add a release note so hopefully this clarifies it a bit.

{{< relnote >}}

## Changes

- **Fixed** an issue where the Istio operator prunes all resources created by the operator, including itself. Now the operator will only remove resources belonging to the custom resource. ([Issue #30833](https://github.com/istio/istio/issues/30833))

- **Fixed** an issue ensuring lease duration is always greater than the user configured `RENEW_DEADLINE` for Istio operator manager. ([Issue #27509](https://github.com/istio/istio/issues/27509))

- **Fixed** an issue where a certificate provisioned by sidecar proxy cannot be used by Prometheus. ([Issue #29919](https://github.com/istio/istio/issues/29919))

- **Fixed** an issue that creates an IOP under `istio-system` when installing Istio in another namespace. ([Issue #31517](https://github.com/istio/istio/issues/31517))

- **Fixed** an issue when using `PeerAuthentication` to turn off mTLS while using multi-network. Now non-mTLS endpoints will be removed from cross-network load-balancing endpoints to prevent 500 errors. ([Issue #28798](https://github.com/istio/istio/issues/28798))

> - **Fixed** istiod never becoming ready when it fails to read resources from clusters configured via remote secrets.

Reviewer 3: ```suggestion

- **Fixed** istiod never becoming ready when it fails to read resources from clusters configured via remote secrets.
```
